rust/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -1,19 +1,47 @@
+use std::{str::FromStr, sync::Arc};
+
 use async_trait::async_trait;
 
 use hyperlane_core::{
     BlockInfo, ChainResult, HyperlaneChain, HyperlaneDomain, HyperlaneProvider, TxnInfo, H256, U256,
 };
+use solana_sdk::{commitment_config::CommitmentConfig, pubkey::Pubkey};
+
+use crate::{client::RpcClientWithDebug, error::HyperlaneSealevelError, ConnectionConf};
 
 /// A wrapper around a Sealevel provider to get generic blockchain information.
 #[derive(Debug)]
 pub struct SealevelProvider {
     domain: HyperlaneDomain,
+    rpc_client: Arc<RpcClientWithDebug>,
 }
 
 impl SealevelProvider {
     /// Create a new Sealevel provider.
-    pub fn new(domain: HyperlaneDomain) -> Self {
-        SealevelProvider { domain }
+    pub fn new(domain: HyperlaneDomain, conf: &ConnectionConf) -> Self {
+        // Set the `processed` commitment at rpc level
+        let rpc_client = Arc::new(RpcClientWithDebug::new_with_commitment(
+            conf.url.to_string(),
+            CommitmentConfig::processed(),
+        ));
+
+        SealevelProvider { domain, rpc_client }
+    }
+
+    /// Get an rpc client
+    pub fn rpc(&self) -> &RpcClientWithDebug {
+        &self.rpc_client
+    }
+
+    /// Get the balance of an address
+    pub async fn get_balance(&self, address: String) -> ChainResult<U256> {
```

Comment:
I think you need to update the HyperlaneProvider::get_balance function on line 77 to call this, or to just move this fn impl to line 77